src/io_threads.c

```diff
@@ -218,7 +218,7 @@ static void *IOThreadMain(void *myid) {
     snprintf(thdname, sizeof(thdname), "io_thd_%ld", id);
     valkey_set_thread_title(thdname);
     serverSetCpuAffinity(server.server_cpulist);
-    makeThreadKillable();
+    pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, NULL);
```

Comment:
nit: suggest also explicitly set pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED, NULL). although it is the default (PTHREAD_CANCEL_ENABLE is also the default) makews more sense to explicitly set both defaults or non.